libs/dyn/pattern_trie_test.go

```diff
@@ -0,0 +1,191 @@
+package dyn_test
+
+import (
+	"testing"
+
+	"github.com/databricks/cli/libs/dyn"
+	assert "github.com/databricks/cli/libs/dyn/dynassert"
+)
+
+func TestPatternTrie_SearchPath(t *testing.T) {
+	tests := []struct {
+		name        string
+		pattern     string
+		expected    []string
+		notExpected []string
+	}{
+		{
+			name:        "empty pattern",
+			pattern:     "",
+			expected:    []string{""},
+			notExpected: []string{"foo"},
```

Comment:
nit: expected / notExpected make it sound like we return all matching patterns, whereas we just check if they match. mustMatch / mustNotMatch would better explain that.